bin/optuna_multiple_gpu.py

```diff
@@ -0,0 +1,56 @@
+import os
+import optuna
+import hydra
+from omegaconf import DictConfig
+
+from bliss.hyperparameter import SleepTune
+
+
+@hydra.main(config_path="../config", config_name="config")
+def main(cfg: DictConfig):
+
+    # where to save the checkpoint
+    model_dir = os.getcwd()
+
+    # which metric to monitor
+    monitor = "val_loss"
+
+    # set up pruner
+    pruner = optuna.pruners.MedianPruner(n_warmup_steps=10)
+
+    # how many gpus do we need
+    n_gpu = 3
+
+    # set up the config for the training
+    cfg.model.encoder.params.update(
+        {
+            "enc_conv_c": (5, 25, 5),
+            "enc_hidden": (64, 128, 64),
+        }
+    )
+
+    cfg.optimizer.params.update({"lr": (1e-4, 1e-2), "weight_decay": (1e-6, 1e-4)})
+
+    SleepTune(
+        cfg,
+        max_epochs=100,
+        model_dir=model_dir,
+        monitor=monitor,
+        n_batches=4,
+        batch_size=32,
+        direction="minimize",
+        data_seed=10,
+        n_trials=100,
+        time_out=600,
+        gc_after_trial=True,
+        sampler=None,  # use the default sampler of the optuna
+        pruner=pruner,
+        num_gpu=n_gpu,
+        storage="sqlite:///zz.db",
```

Comment:
@ismael-mendoza should all these parameters be stored in hydra config files? i’m not sure how it is typically done

Reply:
Yeah that would be a good improvement, I was thinking we can have a config file for hyper-parameter tuning stuff. Not sure if @zzhaozhe-profolio wants to get into this in this PR but that would be nice

Reply:
@ismael-mendoza sampler and pruner need to be constructed by using optuna object. It's not just a number or string. I'm not sure if the config file could do that?

Reply:
I guess it depends how flexible you want the options to be. If you always want to use the median pruner you can just add the keyword arguments of the median pruner to the config file. If you want to allow for multiple types pruners then it gets more complicated. This might be helpful: https://hydra.cc/docs/patterns/instantiate_objects/overview